File: Examples/ObjectiveC/MoPubInterstitialAdapterExample/MoPubInterstitialAdapterExample/ViewController.m
/*
 * Version for iOS © 2015–2021 YANDEX
 *
 * You may not use this file except in compliance with the License.
 * You may obtain a copy of the License at https://yandex.com/legal/mobileads_sdk_agreement/
 */

#import <MoPubSDK/MoPub.h>
#import "ViewController.h"

// Replace XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX with Ad Unit ID generated at https://app.mopub.com.
static NSString *const kMoPubBlockID = @"XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX";

@interface ViewController () <MPInterstitialAdControllerDelegate>

@property (nonatomic, weak) IBOutlet UIButton *loadButton;
@property (nonatomic, strong) MPInterstitialAdController *interstitial;

@end

@implementation ViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
    [self initializeMoPub];
    self.interstitial = [MPInterstitialAdController interstitialAdControllerForAdUnitId:kMoPubBlockID];
    self.interstitial.delegate = self;
}

- (IBAction)loadAd:(UIButton *)sender
{
    [self.interstitial loadAd];
}

- (IBAction)showAd:(UIButton *)sender
{
    [self.interstitial showFromViewController:self];
}

- (void)initializeMoPub
{
    MPMoPubConfiguration *configuration =
        [[MPMoPubConfiguration alloc] initWithAdUnitIdForAppInitialization:kMoPubBlockID];
    __typeof(self) __weak weakSelf = self;
    [[MoPub sharedInstance] initializeSdkWithConfiguration:configuration completion:^{
        dispatch_async(dispatch_get_main_queue(), ^{
            weakSelf.loadButton.userInteractionEnabled = YES;
        });
    }];
}

#pragma mark - MPInterstitialAdControllerDelegate

- (void)interstitialDidLoadAd:(MPInterstitialAdController *)interstitial
{
    NSLog(@"Interstitial did load ad");
}

- (void)interstitialDidFailToLoadAd:(MPInterstitialAdController *)interstitial withError:(NSError *)error
{
    NSLog(@"Interstitial did fail to load ad with error: %@", error);
}

- (void)interstitialWillAppear:(MPInterstitialAdController *)interstitial
{
    NSLog(@"Interstitial will appear");
}

- (void)interstitialDidAppear:(MPInterstitialAdController *)interstitial;
{
    NSLog(@"Interstitial did appear");
}

- (void)interstitialWillDismiss:(MPInterstitialAdController *)interstitial
{
    NSLog(@"Interstitial will dismiss");
}

- (void)interstitialDidDismiss:(MPInterstitialAdController *)interstitial
{
    NSLog(@"Interstitial did dismiss");
}

- (void)interstitialDidReceiveTapEvent:(MPInterstitialAdController *)interstitial
{
    NSLog(@"Interstitial did receive tap event");
}

- (void)mopubAd:(id<MPMoPubAd>)ad didTrackImpressionWithImpressionData:(MPImpressionData *)impressionData
{
    NSLog(@"Interstitial did track impression");
}

@end

File: Examples/Swift/MoPubInterstitialAdapterExample/MoPubInterstitialAdapterExample/ViewController.swift
/*
 * Version for iOS © 2015–2021 YANDEX
 *
 * You may not use this file except in compliance with the License.
 * You may obtain a copy of the License at https://yandex.com/legal/mobileads_sdk_agreement/
 */

import MoPubSDK

class ViewController: UIViewController {

    // Replace XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX with Ad Unit ID generated at https://app.mopub.com.
    let adUnitID = "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"
    
    var interstitial: MPInterstitialAdController?

    @IBOutlet weak var loadButton: UIButton!
    
    override func viewDidLoad() {
        super.viewDidLoad()

        initializeMoPub()
        interstitial = MPInterstitialAdController(forAdUnitId: adUnitID)
        interstitial?.delegate = self
    }

    @IBAction func loadAd(_ sender: UIButton) {
        interstitial?.loadAd()
    }

    @IBAction func showAd(_ sender: UIButton) {
        interstitial?.show(from: self)
    }

    private func initializeMoPub() {
        let configuration = MPMoPubConfiguration(adUnitIdForAppInitialization: adUnitID)
        MoPub.sharedInstance().initializeSdk(with: configuration) {
            DispatchQueue.main.async { [weak self] in
                self?.loadButton.isUserInteractionEnabled = true
            }
        }
    }
}

extension ViewController: MPInterstitialAdControllerDelegate {

    func interstitialDidLoadAd(_ interstitial: MPInterstitialAdController!) {
        print("Interstitial did load ad")
    }

    func interstitialDidFail(toLoadAd interstitial: MPInterstitialAdController!, withError error: Error!) {
        print("Interstitial did fail to load ad with error: \(String(describing: error))")
    }

    func interstitialWillAppear(_ interstitial: MPInterstitialAdController!) {
        print("Interstitial will appear")
    }

    func interstitialDidAppear(_ interstitial: MPInterstitialAdController!) {
        print("Interstitial did appear")
    }

    func interstitialWillDismiss(_ interstitial: MPInterstitialAdController!) {
        print("Interstitial will sidmiss")
    }

    func interstitialDidDismiss(_ interstitial: MPInterstitialAdController!) {
        print("Interstitial did dismiss")
    }

    func interstitialDidReceiveTapEvent(_ interstitial: MPInterstitialAdController!) {
        print("Interstitial did receive tap event")
    }

    func mopubAd(_ ad: MPMoPubAd, didTrackImpressionWith impressionData: MPImpressionData?) {
        print("Interstitial did track impression")
    }
}
